portal/src/main/webapp/js/src/dashboard/iviz.js

```diff
@@ -2255,6 +2351,165 @@ var iViz = (function(_, $, cbio, QueryByGeneUtil, QueryByGeneTextArea) {
+ * If you want to hide chart, please set the priority to 0. For combination chart, as long as one of the clinical attribute has been set to 0, it will be hidden.
+ *
+ * Currently, we preassigned priority to few charts, but as long as you assign a priority in the database except than 1,
+ * these preassigned priorities will be overwritten.
+ *
+ * The first tier: CANCER_TYPE: 3000, CANCER_TYPE_DETAILED: 2000
+ * The second tier: Overall survival plot: 400 (This is combination of OS_MONTH and OS_STATUS), Disease Free Survival Plot: 300 (This is combination of DFS_MONTH and DFS_STATUS), Mutation Count vs. CNA Scatter Plot: 200
+ * The third tier: Mutated Genes Table: 90, CNA Genes Table: 80, study_id: 70, With Mutation Data Pie Chart: 60, With CNA Data Pie Chart: 50, Mutation Count Bar Chart: 30, CNA Bar Chart: 20
+ * The fourth tier: GENDER: 9, SEX: 9, AGE: 8
+ */
+
+
+'use strict';
+(function(iViz, _) {
+  iViz.priorityManager = (function() {
+    var content = {};
+    var clinicalAttrsPriority = {};
+    var defaultPriority = [0, 0, 0, 1];
+
+    /**
+     * Calculate second tier priority by clinical attribute.
+     * @param {string} id Clinical attribute ID.
+     * @return {number}
+     */
+    function getSecondTierPriority(id) {
+      var priority = 1;
+      if (id) {
+        switch (id) {
+        case 'DFS_SURVIVAL':
+          var _dfsStatus = getForthTierPriority('DFS_STATUS');
+          var _dfsMonths = getForthTierPriority('DFS_MONTHS');
+          if (_dfsStatus === 0 || _dfsMonths === 0) {
+            priority = 0;
+          } else {
+            priority = (_dfsMonths + _dfsStatus ) / 2;
+            priority = priority > 1 ? priority :
+              clinicalAttrsPriority['DFS_SURVIVAL'][1];
+          }
+          break;
+        case 'OS_SURVIVAL':
+          var _osStatus = getForthTierPriority('OS_STATUS');
+          var _osMonths = getForthTierPriority('OS_MONTHS');
+          if (_osStatus === 0 || _osMonths === 0) {
+            priority = 0;
+          } else {
+            priority = (_osStatus + _osMonths ) / 2;
+            priority = priority > 1 ? priority :
+              clinicalAttrsPriority['OS_SURVIVAL'][1];
+          }
+          break;
+        case 'MUT_CNT_VS_CNA':
+          priority = clinicalAttrsPriority['MUT_CNT_VS_CNA'][1];
+          break;
+        }
+      }
+      return priority;
+    }
+
+    /**
+     * Get forth tier priority by clinical attribute.
+     * @param{string} id Clinical attribute ID.
+     * @return {number}
+     */
+    function getForthTierPriority(id) {
+      return clinicalAttrsPriority.hasOwnProperty(id)
+        ? clinicalAttrsPriority[id][3] : 1;
+    }
+
+    /**
+     * Convert priority array to a number
+     * @param {array} priority
+     * @return {number}
+     */
+    content.calcPriorityNumber = function(priority) {
+      return _.isArray(priority) ? parseInt(priority.join('')) : 0;
```

Comment:
Ok, looking at code like... 
```javascript
    function getSecondTierPriority(id) {
      var priority = 1;
      if (id) {
        switch (id) {
        case 'DFS_SURVIVAL':
          var _dfsStatus = getForthTierPriority('DFS_STATUS');
```
...I understand now that some attributes are in fact preassigned to a tier, and all custom configuration for these attributes (in the fourth tier) will be calculated with the preassigned tier weight. E.g. in example above, `DFS_SURVIVAL` is part of second tier. If we add custom configuration for priority of `DFS_SURVIVAL` (fourth tier), e.g. priority = 20, then priority will in fact be 100x20=2,000 and **not** the documented 300+20=320. *This is unexpected, given it does not match the given documentation.*